### PR TITLE
Cu 57u5q6 clark chen

### PR DESCRIPTION
**Pull request by Clark-spec**
First Login Change Password

### CODE REVIEW

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryApiController.java [hunk 1]
```diff
@@ -1,56 +1,76 @@
...
 
+    @PostMapping("/add-lottery-holder")
+    public String addLotteryHolder(
+            @RequestParam("email") String email,
+            @RequestParam("firstName") String firstName,
+            @RequestParam("lastName") String lastName,
+            @RequestParam("middleName") String middleName,
+            @RequestParam("fullName") String fullName,
+            @RequestParam("telephone") String telephone
+    ) {
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryApiController.java:25]: More than 6 param, use requestbody

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryHolderService.java [hunk 2]
```diff
@@ -38,25 +45,63 @@
 
     //add a account
     public String addNewLotteryHolder(JSONObject lotteryHolderInfo){
+
         LotteryHolder lotteryHolder = JSON.toJavaObject(lotteryHolderInfo, LotteryHolder.class);
         String email = lotteryHolder.getEmail();
         LotteryHolder lotteryHolderInDb = lotteryHolderRepository.findByEmailEquals(email);
         if (Objects.isNull(lotteryHolderInDb)){
             String initPassword = RandomUtil.getAlphabet();
+            System.out.println(initPassword);
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryHolderService.java:54]: Delete print

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryHolderService.java [hunk 3]
```diff
@@ -112,5 +156,76 @@ public String getAllLotteryHolder(){
...
+    public String lotteryLogin(String loginData){
+        if (StringUtils.isEmpty(loginData)) {
+            LOGGER.warn("There is no login data information");
+            return AjaxResult.failure(CodeEnum.AUTH_ERROR_ACCOUNT_NO_DATA.getValue(), CodeEnum.AUTH_ERROR_ACCOUNT_NO_DATA.getText());
+        }
+
+        JSONObject loginDataObject = JSON.parseObject(loginData);
+        String email = loginDataObject.getString(JsonKeyConst.EMAIL).toLowerCase();
+        String password = loginDataObject.getString(JsonKeyConst.PASSWORD);
+        System.out.println(password);
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryHolderService.java:169]: delete print

**Review by roushanyyzz — CHANGES_REQUESTED** [review]
Update

**Comment by Clark-spec** [comment]
already changed

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryHolderService.java [hunk 4]
```diff
@@ -46,17 +54,53 @@ public String addNewLotteryHolder(JSONObject lotteryHolderInfo){
...
+            String initPassword = RandomUtil.getAlphabet();
+            String salt = RandomUtil.getAlphabet();
+            String encryptInitPassword = CryptUtil.md5(initPassword);
+            String encryptPassword = CryptUtil.md5(encryptInitPassword + salt);
+            lotteryHolder.setStatus(0);
+            lotteryHolder.setPassword(encryptPassword);
+            lotteryHolder.setSalt(salt);
+            lotteryHolder.setFlags("0");
+            lotteryHolderRepository.save(lotteryHolder);
+            System.err.println(AjaxResult.success("success"));
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryHolderService.java:98]: Delete print

Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryHolderService.java]: done

**Review by roushanyyzz — APPROVED** [review]

**Pull request merged**